src/zcl_aff_writer.clas.abap

```diff
@@ -517,16 +517,15 @@ CLASS zcl_aff_writer IMPLEMENTATION.
   ENDMETHOD.
 
   METHOD get_constant_as_struc.
-    DATA clstype TYPE seoclstype.
-    CALL FUNCTION 'SEO_CLIF_EXISTENCE_CHECK'
+    cl_oo_classname_service=>get_class_category(
       EXPORTING
-        cifkey        = CONV seoclskey( name_of_source )
-      IMPORTING
-        clstype       = clstype
+        clsname            = CONV #( name_of_source )
+      RECEIVING
+        result             = DATA(clstype)
       EXCEPTIONS
-        not_specified = 1
-        not_existing  = 2
-        OTHERS        = 3.
+        class_not_existing = 1
+        OTHERS             = 2 ).
+
```

Comment:
hmm, I'm not sure if this works

later in the code is:
`IF clstype = seoc_clstype_interface.`
but the result of the method call is a SEOCATEGRY, which looks different